src/dataframe/dataframe.jl

```diff
@@ -530,6 +557,56 @@ function Base.setindex!(df::DataFrame,
     return df
 end
 
+function checkdimensions(new_df::AbstractDataFrame,
+                         row_inds::Any,
+                         col_inds::Any)
+    new_rows, new_cols = size(new_df)
+    # note that length(::Number) == 1
+    if (new_rows, new_cols) != (length(row_inds), length(col_inds))
```

Comment:
do we handle the case that `row_inds` or `col_inds` have duplicates?
Also do we avoid calling this method if either of them is `AbstractVector{Bool}`?